src/main/java/com/mykaarma/urlshortener/service/UrlService.java

```diff
@@ -136,20 +146,34 @@ public UrlDetails shortenUrl(String longUrl, String shortUrlDomain, long expiryD
 		}
 
 		String shortUrl = shortUrlDomain + urlPrefix + shortUrlHash;
-		
+
 		shortUrlDetails.setShortUrl(shortUrl);
-		
+
 		try {
 			urlRepository.saveUrl(shortUrlDetails);
+		}
+		catch (ShortUrlDuplicateException e)
+		{
+			log.info(String.format("DuplicateHashKeyException Exception Occurred, retrying creating short url for longUrl=%s businessUUID=%s domainPurpose=%s",longUrl, businessUUID, shortUrlDomain));
+			if(retryCount ==6)
+			{
+				log.error("Failed to create shortUrl even after retries for long url=%s",longUrl);
```

Comment:
@harshit050499 also log the exception in here. Also, please test this log, I don't think log.info will recognize `%s`